extensions/modules/file/src/main/java/org/exist/xquery/modules/file/Sync.java

```diff
@@ -64,7 +66,10 @@
                         Cardinality.EXACTLY_ONE, "The full path or URI to the directory"),
                 new FunctionParameterSequenceType("dateTime", Type.DATE_TIME, 
                         Cardinality.ZERO_OR_ONE, 
-                		"Optional: only resources modified after the given date/time will be synchronized.")
+                        "Optional: only resources modified after the given date/time will be synchronized."),
+                new FunctionParameterSequenceType("prune", Type.BOOLEAN,
+                        Cardinality.ZERO_OR_ONE,
```

Comment:
The Cardinality for this should be EXACTLY_ONE. It doesn't really make sense for this to be optional as it is a boolean value.

Reply:
I've made it "optional" with the single purpose that the API remains backward compatible.
Old style invocations of file:sync(...) will still work.
If the ```prune``` param is made non-optional, then:
1. Old-style invocations of that function will be rendered broken (i.e. will have to be revised)
2. IIRC the previous (3rd) param of the function will also need to become non-optional, as to the best of my understanding, you can't have optional param, followed by non-optional, right?

Reply:
That's not backwards compatible. There is no polymorphism in XQuery. You have to keep the original function signature, and then define additional function signatures with additional arities.

Reply:
Wait, wait, wait.
Then (in order not to break existing code) this extended 'prune' functionality has to come together with a new function, e.g. ```file:sync_ex(...)``` or ```file:sync_prune(...)``` or something.
Is it not so?

Reply:
You don't need a new named function as they are on different arities. But you have to design two signatures, the original signature, and the new signature which has an additional argument. The new signature should then be added to the FileModule.java